spec/models/order_spec.rb

```diff
@@ -0,0 +1,217 @@
+require 'rails_helper'
+
+describe Order do
+
+  let(:order) { FactoryBot.create(:order) }
+  before { ActiveJob::Base.queue_adapter = :test }
+
+
+  describe 'constants' do
+    it "is expected to have time format constant" do
+      expect(Order::TIME_FORMAT).to eq("%I:%M%p")
+    end
+
+    it "is expected to have preparation time constant" do
+      expect(Order::PREPARATION_TIME).to eq(1800)
+    end
+  end
+
+  describe 'included module' do
+    it "is expected to have included token generator" do
+      expect(Order.ancestors.include?(TokenGenerator)).to eq(true)
+    end
+  end
+
+  describe 'delegations' do
+    it "is expected to respond to branch_name" do
+      expect(Order.new).to respond_to(:branch_name)
+    end
+
+    it "is expected to respond to branch_address" do
+      expect(Order.new).to respond_to(:branch_address)
+    end
+
+    it "is expected to respond to branch_contact_number" do
+      expect(Order.new).to respond_to(:branch_contact_number)
+    end
+  end
+
+  describe 'associations' do
+    it { is_expected.to have_one(:charge).dependent(:destroy) }
+    it { is_expected.to belong_to(:cart) }
+    it { is_expected.to belong_to(:user) }
+    it { is_expected.to belong_to(:branch) }
+  end
+
+  describe 'validations' do
+    it { is_expected.to validate_presence_of(:phone_number) }
+    it { is_expected.to validate_presence_of(:pick_up) }
+    it { is_expected.to allow_value('9654208158').for(:phone_number) }
+    it { is_expected.not_to allow_value('965420815').for(:phone_number) }
+
+  end
+
+  describe 'custom validations' do
+    it { is_expected.to callback(:valid_pick_up_time?).before(:validate).if(:pick_up?) }
+    it { is_expected.to callback(:sufficient_preparation_time?).before(:validate).if(:pick_up?) }
+    it { is_expected.to callback(:past_pick_up?).before(:validate).if(:pick_up?) }
+  end
+
+  describe 'callback' do
+    it { is_expected.to callback(:send_confirmation_mail).after(:commit).on(:create) }
+  end
+
+  describe 'scope' do
+    describe 'by_date' do
+      it "is expected to by default return orders created in last 7 days" do
+        expect(Order.by_date).to include(order)
+      end
+
+      it "is expected to return orders created between time given" do
+        expect(Order.by_date(2.month.ago , 1.month.ago)).not_to include(order)
+      end
+    end
+  end
+
+  describe 'Instance Public Methods' do
+    describe 'delivered' do
+      it 'is expected to change the picked status of order' do
+        order_deliver_status = order.picked
```

Comment:
move these lines in a before
In most of the cases it only have the expect statement nothing else